### PR TITLE
Add an empty string check for  topic2tables map

### PR DESCRIPTION
**Pull request by SajanaW**
Signed-off-by: SajanaW <sweerawardhena@confluent.io>

## Problem
- We had a cloud issue recently, where the connector failed to start because topic2table map was an empty string. This was an elusive error. For context, the tasks fail when a hot fix was rolled, and then the second hot fix, which was supposed to fix the bug introduced by the first one, was not able to remediate this connector. The reason was topic2table map was causing the connector not to start up/ it errored out.  But this error just gets logged in the connector and a generic error message was being thrown. After much debugging we realized what is going on.  Error took the form:
It logged this
```
[ERROR] 2021-06-22 22:55:32,614 [connector-thread-lcc-36n2w] com.snowflake.kafka.connector.Utils parseTopicToTableMap -
[SF_KAFKA_CONNECTOR] Invalid snowflake.topic2table.map config format:
```

But threw this:

```
[SF_KAFKA_CONNECTOR] Exception: Invalid input connector configuration
[SF_KAFKA_CONNECTOR] Error Code: 0001
[SF_KAFKA_CONNECTOR] Detail: input kafka connector configuration is null, missing required values, or wrong input value
	at com.snowflake.kafka.connector.internal.SnowflakeErrors.getException(SnowflakeErrors.java:281)
	at com.snowflake.kafka.connector.internal.SnowflakeErrors.getException(SnowflakeErrors.java:248)
	at com.snowflake.kafka.connector.Utils.validateConfig(Utils.java:494)
	at com.snowflake.kafka.connector.SnowflakeSinkConnector.start(SnowflakeSinkConnector.java:86)
	at org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:185)
	at org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:210)
	at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:349)
	at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:332)
	at org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:141)
	at org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:118)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
```
- In the connector start method, there is a validation check for topic2table, which fails the connector if is is "". For some reason our lower level configs materialize this config as an empty string. 

## Solution
- For the moment, we are trying to patch by explicity checking if empty string and skipping the validation.

In the future we have to move the validation checks from the SnowFlake start method to the validate method

##### Does this solution apply anywhere else?
- [ ] yes
- [ x] no

##### Testing done:
- [ ] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan

 
Hotfix stag and test

### CODE REVIEW

**Comment by aakashnshah** [comment]
> For the moment, we are trying to patch by explicity checking if empty string and skipping the validation.
In the future we have to move the validation checks from the SnowFlake start method to the validate method

Are you not doing this check in config validation and not `start`?

File: src/main/java/com/snowflake/kafka/connector/Utils.java [hunk 1]
```diff
@@ -415,7 +415,8 @@ static String validateConfig(Map<String, String> config) {
     }
 
     if (config.containsKey(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MAP)
-        && parseTopicToTableMap(config.get(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MAP))
+            && !config.get(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MAP).equals("")
```
Comment (ddasarathan) [inline src/main/java/com/snowflake/kafka/connector/Utils.java:418]: nit: replace with https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#isEmpty() ?
`config.get(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MAP).isEmpty()`

**Review by ddasarathan — APPROVED** [review]
Minor comment, Otherwise LGTM

**Review by ncliang — APPROVED** [review]
Thanks @SajanaW , LGTM. @liukrimhrim , is the patch for `behavior.on.null` in upstream `v1.5.5` release? If so, do we want to refresh snowflake to `1.5.5` while we are at it?

**Comment by liukrimhrim** [comment]
@SajanaW @ncliang I think so. The new prop was introduced since 1.5.4, so 1.5.5 should contain it as well

**Pull request merged**